src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -365,20 +394,34 @@ def run(self, batch_inputs):
             ),
         ):
             try:
+
+                base_args = (self._inputs_queue, self._line_timeout_sec, result_list)
+
+                args_list = [
+                    base_args + (i, self._input_queues[i], self._output_queues[i])
+                    for i in range(self._n_process)
+                ]
+
                 # The variable 'async_result' here is not the actual result of the batch run
                 # but an AsyncResult object that can be used to check if the execution are finished
                 # The actual results of the batch run are stored in 'result_list'
-                async_result = self._pool.starmap_async(
-                    self._timeout_process_wrapper,
-                    [
-                        (run_start_time, self._inputs_queue, self._line_timeout_sec, result_list)
-                        for _ in range(self._n_process)
-                    ],
-                )
+                async_result = self._pool.starmap_async(self._timeout_process_wrapper, args_list)
+
                 try:
+                    # Only log when the number of results changes to avoid duplicate logging.
+                    last_log_count = 0
                     # Wait for batch run to complete or KeyboardInterrupt
                     while not async_result.ready():
-                        # Check every 1 second
+                        current_result_count = len(result_list)
+                        if current_result_count != last_log_count:
+                            log_progress(
+                                run_start_time=run_start_time,
```

Comment:
Do we have test case to cover the logs here？

Reply:
Added